README.md

```diff
@@ -269,58 +271,57 @@ onDialNumber(data) {
 </p>
 </details>
 
-### Running the demo Calling Extension Widget project
-
-The calling extensions are enabled for any portal that is starter or above.
-
-#### Run the demo widget project
-
-```shell
-# install npm build dependencies and start the demo project
-cd /demo
-npm i
-npm start
-```
-
-Load the demo page in chrome and accept the invalid cert exception
-
-#### Add local storage override for calling extensions
-
-Add the following localstorage override for testing purposes -
+## Typical message flow between the call widget and HubSpot
 
-```js
-localStorage.setItem(
-  "LocalSettings:Sales:CallingExtensions",
-  '{"name": "Demo widget", "url": "https://localhost:9025/"}'
-);
-```
+### Initializing the call widget
 
-#### Navigate to a contacts/company page and launch call
+The following messages are exchanged when a call widget is instantiated.
+![Image description](./docs/images/InitializeCallWidgetIFrame.png)
+Once the widget iFrame is created, Hubspot send the SYNC message to the widget after iFrame is loaded and repetedly send this message until it receives the SYNC_ACK response from the widget. If the SYNC_ACK response isn't received within 30 seconds, the widget is marked as failed. Note that sending SYNC/SYNC_ACK messages are handled by the framework. Once the widget and host page are synchronized, the frameworks triggers the ready event.
 
-The calling extension demo widget should load inside an iFrame.
+The call widget should waits for the ready event from the framework and send the initialized event to HubSpot. At this point, the messages can be exchanged between the call widget and HubSpot.
 
-## Typical message flow between the call widget and HubSpot
+### Outbound call
 
-### Loading the call widget
+The following messages are exchanged when user initiates a call.
+![Image description](./docs/images/OutboundCallSequenceDiagram.png)
+Hubspot ensures the call widget is logged in before sending in a dial number event - if the widget is not logged in, an alert is shown in HubSpot's UI.
 
-The following messages are exchanged when a call widget is instantiated -
+Here is description of events:
 
-1. [HubSpot] Sends a SYNC message
-2. [SoftPhone] Sends SYNC_ACK message
-3. [SoftPhone] Sends INITIALIZED message with login state and optionally widget size
+1. **Dial number** - HubSpot sends the dial number event.
+2. **Outbound call started** - Widget notifies HubSpot when the call is started.
+3. **Create engagement** - HubSpot creates an engagement with minimum information if requested by the widget.
+4. **Engagement created** - HubSpot creates an engagement.
+5. **Engagement created** - HubSpot sends the engagementId to the widget.
```

Comment:
Should there be 2 `**Engagement created**` here?

Reply:
I will revisit the names.